### ISSUE TITLE
Cloud-init is disabled in Ubuntu 24.04 images with `vm_guest_os_cloudinit=true`

### ISSUE THREAD

**Issue by sepsh** [vmware-samples/packer-examples-for-vsphere#939]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

develop

### VMware vSphere

7.0.3

### HashiCorp Packer

1.11.0

### HashiCorp Packer Plugin for VMware vSphere

1.3.0

### Guest Operating System

Ubuntu Server 24.04 LTS

### Environment Details

- Built with `vm_guest_os_cloudinit=true`
- Packer host: Packer `1.11.0`
- Packer host: Packer vSphere plugin `1.3.0`
- Packer host: Python `3.12.3`
- Packer host: Ansible `10.0.1`
- Terraform host: Terraform `1.6.0`
- Terraform host: Terraform vSphere provider `2.5.1`
- Packer, Terraform, Ansible host: `Kubuntu 22.04`
- vSphere version `7.0.3`
- Guest OS `ubuntu-24.04-live-server-amd64.iso`
- Guest OS open-vm-tools `2:12.3.5-5build3`
- Guest OS cloud-init `24.1.3-0ubuntu3.3`

### Description

Hi. When I tried building an image from Ubuntu Server 24.04 LTS and cloning it with Terraform, I noticed that the user configuration, hostname and network configuration and other stuff set in cloud-init templates, was not applied.

Then I ran:
```bash
cloud-init status
```
I got `disabled`

### Expected Behavior

Cloud-init beeing enabled

### Actual Behavior

Cloud-init is not enabled

### Steps to Reproduce

- Build an image using `builds/linux/ubuntu/24-04-lts` and set `vm_guest_os_cloudinit=true`
- Provision a Virtual Machine using `terraform/vsphere-virtual-machine/content-library-ovf-linux-cloud-init`
- Login into the VM and Check if any config from `terraform/vsphere-virtual-machine/content-library-ovf-linux-cloud-init/metadata.yml` or `terraform/vsphere-virtual-machine/content-library-ovf-linux-cloud-init/userdata.yml` was applied
- run `cloud-init status`
- Check if the file `/etc/cloud/cloud-init.disabled` is present.
- Ensure that by checking `cat /var/lib/cloud/instances/iid-datasource-none/cloud-config.txt` and see if that looks like this: 
![Screenshot_20240618_192102](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/56169910/a2c20d17-e0fa-4173-8f78-5e0f1ffb8e4d)

### Screenshots

![Screenshot_20240618_191253](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/56169910/3cd2977b-e86e-4671-a074-fbed344b8ff4)

**Comment by tenthirtyam** [comment]
@burnsjared0415 can you take a look at this one?

**Comment by sepsh** [comment]
One way to fix that could be to add a task in `packer/ansible/roles/configure/tasks` right before
https://github.com/vmware-samples/packer-examples-for-vsphere/blob/d2fde04a56867801bdaeb550e8aed3bf8e62af3f/ansible/roles/configure/tasks/debian.yml#L51

Just adding this could fix it maybe,
```yaml
- file:
    path: "/etc/cloud/cloud-init.disabled"
    state: absent
```

but I'm not sure if it's the best way since this document exist: 
https://cloudinit.readthedocs.io/en/latest/howto/disable_cloud_init.html#method-2-kernel-command-line

**Comment by burnsjared0415** [comment]
will take look this afternoon and look at the suggestion and test.

**Comment by sepsh** [comment]
thanks a lot

**Comment by burnsjared0415** [comment]
@sepsh pull request is in to fix the issue, if you want to pull down the fix and see if it works?

**Comment by tenthirtyam** [comment]
Addressed in #940.

**Comment by sepsh** [comment]
Hi, Sorry for my delay.
That solved my problem, Thanks a lot.
